### PR TITLE
fix(plugin-patch): diffFolders returned silently when 'git' failed

### LINKED ISSUE #2331: [Bug] patch-commit does not return any output/error
- [x] I'd be willing to implement a fix

**Describe the bug**

Running `yarn patch-commit /path/to/generated/dir` returns without output nor does it generate any file.

**To Reproduce**

```
yarn init --yes
yarn add slate-react
yarn patch slate-react
# make some changes to the package in the output directory
yarn patch-commit $dir
```

This returns no output.
For some reason, it does not reproduce in sherlock (Sherlock gives output), but I can consistently reproduce locally. 

Tried latest version and built from master.
Tried Node 12, 14 and 15.

<details>
  <summary>Reproduction</summary>

  ```js repro
const {writeFile} = require('fs').promises
await writeFile(`./.yarnrc.yml`, 'nodeLinker: node-modules')

await yarn(`init`, `--yes`);

await yarn(`add`, `slate-react`);

const output1 = await yarn(`patch`, `slate-react`);
const [, line] = output1.split('\n')
const [, dir] = line.split('folder: ')

await writeFile(`${dir}/dist/index.es.js`, '// some changes')

const output2 = await yarn(`patch-commit`, dir);
expect(output2).not.toBe('')
  ```
</details>

**Screenshots**

Screen Shot 2021-01-04 at 9 38 07 PM

**Environment if relevant (please complete the following information):**

 - OS: OSX 10.15.7 [e.g. OSX, Linux, Windows, ...]
 - Node version: Tried Node 12, 14 and 15. [e.g. 8.15.0, 10.15.1, ...]
 - Yarn version: Tried latest and from master [e.g. 2.4.0, ...]

**Additional context**

As an aside, the output from the command that sherlock gives is not consistent with the documentation, which states:

> This will turn the folder passed in parameter into a patchfile

It actually doesn't generate a patchfile, but outputs it into stdout.

### PR DESCRIPTION
**Pull request by niieani**
**What's the problem this PR addresses?**

Running `yarn patch-commit /path/to/generated/dir` returned without output nor generates any file when an incompatible `git` was present in the system.

The `diffFolders` function runs `git` and disregarded the return status `code` and `stderr`, which means it resolved with an empty value of `stdout`.

The default `git` that's included with OS X does not support one of the arguments passed to it. Example:

```sh
❯ git diff --src-prefix=a/ --dst-prefix=b/ --ignore-cr-at-eol --full-index --no-index /private/.../xfs-adf9c270 /private/.../xfs-e4548122
fatal: invalid diff option/value: --ignore-cr-at-eol
```

Berry thinks an empty string is the result of the diff, prints it to stdout and incorrectly returns status 0.

Closes #2331

**How did you fix it?**

Added a helpful & actionable error message.

I've also checked for other usages of `execUtils.execvp` without `strict: true`. There is one other that also ignores the status code, but I believe it's intentional - it's in the init command (runs `git init` at the very end).

Also made the description of what the command does a bit clearer (it doesn't actually generate any files, it only outputs it to stdout).

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by niieani** [comment]
@arcanis not sure what's going on with the `yarn version check` in the GH Action here. It's complaining that I haven't updated the `.yarn/versions`, but I have. When I run `yarn version check --interactive` everything is good to go and configured. 

Never done any releases in a monorepo, am I doing something wrong?

File: .yarn/versions/52395a7b.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+releases:
+  "@yarnpkg/cli": patch
+  "@yarnpkg/plugin-compat": patch
```
Comment (merceyz) [inline .yarn/versions/52395a7b.yml:3]: `@yarnpkg/plugin-compat` doesn't need a release for this change

Comment (niieani) [reply .yarn/versions/52395a7b.yml]: Fixed.

**Comment by arcanis** [comment]
Thanks @niieani!

**Pull request merged**